pkg/datapath/linux/config/config.go

```diff
@@ -863,7 +863,7 @@ func (h *HeaderfileWriter) writeTemplateConfig(fw *bufio.Writer, e datapath.Endp
 		fmt.Fprintf(fw, "#define USE_BPF_PROG_FOR_INGRESS_POLICY 1\n")
 	}
 
-	if option.Config.ForceLocalPolicyEvalAtSource {
+	if option.Config.ForceLocalPolicyEvalAtSource || (option.Config.EnableEndpointRoutes && !option.Config.EnableHostLegacyRouting) {
```

Comment:
> IIUC, this is okay and won't end up causing #14657 because with BPF Host Routing, we'll skip the second policy enforcement. Is that correct?

Yes, we jump right into the pod namespace.

> I think I would prefer to have that bit of logic in the datapath (so using the two corresponding macros) with a big comment to explain why we're doing that and why it's fine.

```
/*
 * When the host routing is enabled we need to check policies at source.
 * Otherwise, as the skb will be delivered directly to pod's namespace,
 * the ingress policy (the cil_to_container BPF program) will be bypassed.
 */
#if defined(ENABLE_ENDPOINT_ROUTES) && defined(ENABLE_HOST_ROUTING)
#  ifndef FORCE_LOCAL_POLICY_EVAL_AT_SOURCE
#  define FORCE_LOCAL_POLICY_EVAL_AT_SOURCE
#  endif
#endif
```
Does this work for you @pchaigno ?